Reqnroll.VisualStudio/Editor/Commands/FindUnusedStepDefinitionsCommand.cs

```diff
@@ -0,0 +1,206 @@
+                IdeScope.Actions.ShowProblem(
+                    "Unable to find step definition usages: could not find any Reqnroll project with feature files.");
+                return true;
+            }
+
+            var asyncContextMenu = IdeScope.Actions.ShowAsyncContextMenu(PopupHeader);
+            Task.Run(
+                () => FindUnusedStepDefinitionsInProjectsAsync(reqnrollTestProjects, fileName, triggerPoint, asyncContextMenu,
+                    asyncContextMenu.CancellationToken), asyncContextMenu.CancellationToken);
+            return true;
+        }
+
+        private async Task FindUnusedStepDefinitionsInProjectsAsync(IProjectScope[] reqnrollTestProjects, string fileName,
+            SnapshotPoint triggerPoint, IAsyncContextMenu asyncContextMenu, CancellationToken cancellationToken)
+        {
+            var summary = new UnusedStepDefinitionSummary();
+
+            try
+            {
+                await FindUsagesInternalAsync(reqnrollTestProjects, fileName, triggerPoint, asyncContextMenu,
+                    cancellationToken, summary);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogException(MonitoringService, ex);
+                summary.WasError = true;
+            }
+
+            if (summary.WasError)
+                asyncContextMenu.AddItems(new ContextMenuItem("Could not complete find operation because of an error"));
+            else if (summary.UnusedStepDefinitions == 0)
+                asyncContextMenu.AddItems(
+                    new ContextMenuItem("Could not find any unused step definitions at the current position"));
+            //TODO: determine if this required/needed
+            //else if (summary.UsagesFound == 0) asyncContextMenu.AddItems(new ContextMenuItem("Could not find any usage"));
+
+            //TODO: modify monitoring to include finding unused step definitions
+            //MonitoringService.MonitorCommandFindStepDefinitionUsages(summary.UsagesFound,
+            //    cancellationToken.IsCancellationRequested);
+            if (cancellationToken.IsCancellationRequested)
+                Logger.LogVerbose("Finding unused step definitions cancelled");
+            else
+                Logger.LogInfo($"Found {summary.UnusedStepDefinitions} unused step definitions in {summary.ScannedFeatureFiles} feature files");
+            asyncContextMenu.Complete();
+            Finished.Set();
+        }
+        private async Task FindUsagesInternalAsync(IProjectScope[] reqnrollTestProjects, string fileName,
+    SnapshotPoint triggerPoint, IAsyncContextMenu asyncContextMenu, CancellationToken cancellationToken,
+    UnusedStepDefinitionSummary summary)
+        {
+            foreach (var project in reqnrollTestProjects)
+            {
+                if (cancellationToken.IsCancellationRequested)
+                    break;
+
+                var stepDefinitions = await GetStepDefinitionsAsync(project, fileName, triggerPoint);
```

Comment:
We do not need to collect step definitions per file, because we can use the step definitions for the project (see comment below). So this few lines and the related methods (`GetStepDefinitionsAsync`, `GetStepDefinitions`) are not needed. Consequently you will not need the `fileName` parameter either.

Reply:
Refactored per your suggestion.